src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -692,3 +588,112 @@ def update_connectedk8s(cmd, instance, tags=None):
+
+    # Removing quotes from kubeconfig path. This is necessary for windows OS.
+    trim_kube_config(kube_config)
+
+    # Escaping comma, forward slash present in https proxy urls, needed for helm params.
+    https_proxy = escape_proxy_settings(https_proxy)
+
+    # Escaping comma, forward slash present in http proxy urls, needed for helm params.
+    http_proxy = escape_proxy_settings(http_proxy)
+
+    # Escaping comma, forward slash present in no proxy urls, needed for helm params.
+    no_proxy = escape_proxy_settings(no_proxy)
+
+    # Loading the kubeconfig file in kubernetes client configuration
+    try:
+        config.load_kube_config(config_file=kube_config, context=kube_context)
+    except Exception as e:
+        telemetry.set_user_fault()
+        telemetry.set_exception(exception=e, fault_type=consts.Load_Kubeconfig_Fault_Type,
+                                summary='Problem loading the kubeconfig file')
+        raise CLIError("Problem loading the kubeconfig file." + str(e))
+    configuration = kube_client.Configuration()
+
+    # Checking the connection to kubernetes cluster.
+    # This check was added to avoid large timeouts when connecting to AAD Enabled AKS clusters
+    # if the user had not logged in.
+    check_kube_connection(configuration)
+
+    # Get kubernetes cluster info for telemetry
+    kubernetes_version = get_server_version(configuration)
+    kubernetes_distro = get_kubernetes_distro(configuration)
+
+    kubernetes_properties = {
+        'Context.Default.AzureCLI.KubernetesVersion': kubernetes_version,
+        'Context.Default.AzureCLI.KubernetesDistro': kubernetes_distro
+    }
+    telemetry.add_extension_event('connectedk8s', kubernetes_properties)
+
+    # Checking helm installation
+    check_helm_install(kube_config, kube_context)
+
+    # Check helm version
+    helm_version = check_helm_version(kube_config, kube_context)
+    telemetry.add_extension_event('connectedk8s', {'Context.Default.AzureCLI.HelmVersion': helm_version})
+
+    # Check whether Connected Cluster is present
+    if not connected_cluster_exists(client, resource_group_name, cluster_name):
+        telemetry.set_user_fault()
+        telemetry.set_exception(exception='The connected cluster resource does not exist', fault_type=consts.Resource_Does_Not_Exist_Fault_Type,
+                                summary='Connected cluster resource does not exist')
+        raise CLIError("The connected cluster resource {} does not exist ".format(cluster_name) +
+                       "in the resource group {} ".format(resource_group_name) +
+                       "Please onboard the connected cluster using: az connectedk8s connect -n <connected-cluster-name> -g <resource-group-name>")
+
+    # Fetch Connected Cluster for agent version
+    connected_cluster = get_connectedk8s(cmd, client, resource_group_name, cluster_name)
+
+    # Adding helm repo
+    utils.add_helm_repo(kube_config, kube_context)
```

Comment:
Ideally this should be update_helm_repo instead of add_helm_repo. Does running "helm repo add" second time updates the repo?

Reply:
i think it should be add as the repo might have been deleted or if user is using a different machine to update where repo is not there

Reply:
add does update too i believe